applications/sintering/include/pf-applications/sintering/projection.h

```diff
@@ -0,0 +1,210 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2024 by the hpsint authors
+//
+// This file is part of the hpsint library.
+//
+// The hpsint library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 3.0 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.MD at
+// the top level directory of hpsint.
+//
+// ---------------------------------------------------------------------
+
+#pragma once
+
+#include <deal.II/base/geometry_info.h>
+#include <deal.II/base/mpi_large_count.h>
+#include <deal.II/base/table_handler.h>
+
+#include <deal.II/fe/fe_dgq.h>
+
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+#include <deal.II/matrix_free/matrix_free.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include <pf-applications/base/data.h>
+
+#include <pf-applications/sintering/advection.h>
+#include <pf-applications/sintering/operator_sintering_data.h>
+#include <pf-applications/sintering/postprocessors.h>
+
+#include <pf-applications/grain_tracker/distributed_stitching.h>
+#include <pf-applications/grain_tracker/tracker.h>
+#include <pf-applications/grid/grid_tools.h>
+
+namespace Sintering
+{
+  namespace Postprocessors
+  {
+    template <int dim, typename Number>
+    struct StateData
+    {
+      using BlockVector = std::vector<Vector<Number>>;
+
+      Triangulation<dim> tria;
+      DoFHandler<dim>    dof_handler;
+      BlockVector        solution;
+      FE_DGQ<dim>        fe_dg{1};
+
+      StateData()
+        : dof_handler(tria)
+      {}
+
+      StateData(unsigned int n)
+        : dof_handler(tria)
+        , solution(n)
+      {}
+
+      StateData(const StateData<dim, Number> &state) = delete;
+    };
+
+    template <typename VectorType, int dim = 3>
+    std::unique_ptr<StateData<dim - 1, typename VectorType::value_type>>
+    build_projection(const DoFHandler<dim> &background_dof_handler,
```

Comment:
Do you know: https://www.dealii.org/developer/doxygen/deal.II/classDataOutResample.html?

Reply:
I checked that, but these were my considerations to end up with this current solution:
- If I understood correctly, this class needs to have the grid, on which I would like to project the output, already created beforehand. I could have created, let's say, a structured 2D grid at a given location within the domain bounding box, that would work, but I wanted to have the same cells in the output.
- Furthermore, I wanted to have an explicitly created projected solution vector, since later it will be used for contours VTK and TEX output.
- I will also (not yet implemented) attach to each cell of the projected triangulation the `global_active_cell_index()` of its parent to `material_id` field of the child. This is in order to transfer properly information about grain indices and order params correctly onto the projected 2D grid.

In general, if there are tools available in deal.II that could simplify and generalize the algo (at the moment it introduces many assumptions), I woild happily take them.